tron/utils/authentication.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import os
+from functools import lru_cache
+from functools import partial
+from typing import cast
+from typing import List
+from typing import Optional
+from typing import Set
+
+import staticconf  # type: ignore
+from task_processing.plugins.kubernetes.types import ProjectedSAVolume
+
+from tron.config.static_config import NAMESPACE as TRON_NAMESPACE
+
+
+log = logging.getLogger(__name__)
+TOKEN_VOLUME_CONF_NAMESPACE = "service_account_token"
+AUTH_SERVICES_CONF_NAMESPACE = "authenticating_services"
+DEFAULT_SOA_PATH = "/nail/etc/services"
+DEFAULT_PAASTA_CONFIG_PATH = "/etc/paasta"
+CACHED_AUTHENTICATING_SERVICES = None
+
+
+@lru_cache
+def _get_config_watcher(filepath: str, namespace: str) -> Optional[staticconf.ConfigurationWatcher]:
+    """Load configuration file and return a watcher for it
+
+    :param str filepath: path to JSON/YAML configuration
+    """
+    watcher = None
+    loader_class = staticconf.JSONConfiguration if filepath.endswith(".json") else staticconf.YamlConfiguration
+    loader = partial(loader_class, filepath, namespace=namespace, flatten=False)
+    reloader = (staticconf.config.ReloadCallbackChain(namespace),)
+    try:
+        loader()
+        watcher = staticconf.ConfigurationWatcher(loader, filepath, min_interval=10, reloader=reloader)
+    except Exception as e:
+        # soft failing, as authz features are currently optional
+        log.warning(f"Failed loading {filepath}: {e}")
+    return watcher
+
+
+def get_authenticating_services() -> Set[str]:
+    """Load list of services participating in authenticated traffic
+
+    :param str soa_dir: SOA configuration directory
+    :return: set of service names
+    """
+    global CACHED_AUTHENTICATING_SERVICES
+    soa_path = staticconf.read_string("soa_path", namespace=TRON_NAMESPACE, default=DEFAULT_SOA_PATH)
```

Comment:
just curious: any scenarios where we'd want to change this path just for some tron masters?

Reply:
(and same with `DEFAULT_PAASTA_CONFIG_PATH`)

Reply:
No. It's just fear of "we changed that file location from some $reason, but Tron is currently pinned, and we are are stuck".